### PR TITLE
ingestByTimeInterval

### PR DESCRIPTION
**Pull request by ohadbitt**
ingest by time interval

### CODE REVIEW

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java [hunk 1]
```diff
@@ -14,8 +14,6 @@
 public class KustoSinkConfig extends AbstractConfig {
     // TODO: this might need to be per kusto cluster...
     static final String KUSTO_URL = "kusto.url";
-    static final String KUSTO_DB = "kusto.db";
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java]: but you never use it

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/GZIPFileWriter.java [hunk 2]
```diff
@@ -55,6 +81,10 @@ public synchronized void write(byte[] data) throws IOException {
 
         if ((currentFile.rawBytes + data.length) > fileThreshold) {
             rotate();
+            timer.cancel();
+            timer.purge();
+            timer = new Timer(true);
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/GZIPFileWriter.java]: from what i saw the only way to do it is by cancelling the old one and recreating.

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (danield137) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java]: True 😲

**Comment by danield137** [comment]
Can we add a test for time based file rolling?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/GZIPFileWriter.java [hunk 3]
```diff
@@ -33,13 +37,35 @@
...
         this.fileThreshold = fileThreshold;
         this.onRollCallback = onRollCallback;
+        this.flushInterval = flushInterval;
+        this.timer = new Timer(true);
+
+        resetFlushByTime();
+    }
+
+    private void resetFlushByTime() {
+        timer.schedule(new TimerTask() {
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/GZIPFileWriter.java]: you can call run and close on it . but once its closed you can't reuse it - that is why we need to create a new task everytime

**Pull request merged**